### PR TITLE
plugins/otter: add warning if treesitter highlight not enabled

### PR DESCRIPTION
**Pull request by khaneliman**
Follow up to https://github.com/nix-community/nixvim/issues/1854

### CODE REVIEW

File: plugins/languages/otter.nix
Comment (MattSturgeon) [inline plugins/languages/otter.nix:88]:
```
error: expected a Boolean but found null: null
at /nix/store/l6ml4gkff7nzqfqwz4j86a696cnfpcql-source/plugins/languages/otter.nix:88:43:
87| extraConfig = cfg: {
88| warnings = lib.optional (!(cfg.enable -> config.plugins.treesitter.settings.highlight.enable)) ''
| ^
```

Since the `highlight.enable` settings option is nullable, we need to handle that.

Note: if you did `enable == true` then `nix fmt` would have "fixed" that for you, leading to this error.

These would be true when enable is null: `enable != null -> enable` or `isBool enable -> enable`

But I think we actually want false when null, so lets go with: `enable != null && enable` or `isBool enable && enable`.

Probably best to have a small helper in a let block: `isTrue = v: isBool v && v`, or maybe we can use [these helpers](https://github.com/nix-community/nixvim/blob/216d64c158da5523d5b3db0895e1345175c21502/lib/utils.nix#L106-L110)?

Comment (traxys) [reply plugins/languages/otter.nix:88]: Could we only check for `null` and not `false`? Maybe the user will do _something else_ to enable it, and as such we don't want to warn him each time?

File: plugins/languages/otter.nix
Comment (MattSturgeon) [inline plugins/languages/otter.nix:89]:
```suggestion
      lib.optional (cfg.enable && config.plugins.treesitter.settings.highlight.enable == null)
```

Using `->` here would show the warning whenever `cfg.enable == false`

Comment (khaneliman) [reply plugins/languages/otter.nix:89]: That's not what i'm observing in the local tests. If I set otter to disabled in the tests  I dont get the warning.

Comment (MattSturgeon) [reply plugins/languages/otter.nix:89]: That's probably because `highlight.enable == null` when you're testing?

`a -> b` is equivalent to `!a || b`. Here I'm pretty sure we want to show the warning only when _both_ `enable` is true and `highlight.enable` is null.

Comment (khaneliman) [reply plugins/languages/otter.nix:89]: I added ``plugins.treesitter.settings.highlight.enable = true; to the default test and then set otter enable = false.
``

Comment (khaneliman) [reply plugins/languages/otter.nix:89]: I tried a few different combinations in the otter default test. They all didn't result in a warning. Is the mkNeovimPlugin doing something to prevent it?

- `plugins.treesitter.settings.highlight.enable = true;` and `otter.enable = false`
- `plugins.treesitter.settings.highlight.enable = false;` and `otter.enable = false`
- `plugins.treesitter.settings.highlight.enable = null;` and `otter.enable = false`

Comment (khaneliman) [reply plugins/languages/otter.nix:89]: That said, I see the same behavior doing it with -> and && so I can change it, just wanted to point out I wasn't seeing the behavior mentioned.

**Comment by MattSturgeon** [comment]
Should we also be checking `plugins.treesitter.enable`? I guess there'd be no way to disable the warning if the user doesn't want to use our treesitter module though.

**Review by MattSturgeon — APPROVED** [review]
LGTM, thanks!

#1926 is in the merge queue atm, if I queue this then mergify will rebase it (in Gaétan's name) before merging.

I don't mind waiting for you to rebase after the other PR is merged before queueing.

**Comment by MattSturgeon** [comment]
@mergifyio queue

**Pull request merged**
